python_modules/dagster/dagster/core/storage/event_log/in_memory.py

```diff
@@ -84,21 +80,37 @@ def get_logs_for_run(
             else None
         )
 
-        cursor = cursor + 1
+        if cursor is None:
+            offset = 0
+        else:
+            try:
+                offset = int(cursor) + 1
+            except ValueError:
+                offset = 0
+
         if of_types:
             events = list(
                 filter(
-                    lambda r: r.is_dagster_event and r.dagster_event.event_type_value in of_types,
-                    self._logs[run_id][cursor:],
+                    lambda r: r.is_dagster_event
+                    and r.dagster_event.event_type_value in cast(Set[str], of_types),
+                    self._logs[run_id],
                 )
             )
+            events = events[offset:]
         else:
-            events = self._logs[run_id][cursor:]
+            events = self._logs[run_id][offset:]
 
         if limit:
             events = events[:limit]
 
-        return events
+        return EventLogConnection(
+            records=[
+                EventLogRecord(storage_id=event_id + offset, event_log_entry=event)
+                for event_id, event in enumerate(events)
+            ],
+            cursor=str(offset + len(events)),
```

Comment:
lets set up a prefix scheme out of the gate `offset:<N>` 
maybe base64 for some obfuscation to really hit home shouldn't be manually creating